### PR TITLE
[NativeAOT-LLVM] Integrate the JS implementation for `dotnet_browser_entropy`

### LINKED ISSUE #2165: [NativeAOT-LLVM] Runtime tests generate a significant number of warnings
Let's get at least `HelloWasm` clean, eleven warnings is too high a number to ignore.

Some of these are upstream problems we've picked up, but most should be actionable directly.

Current list of warnings:

1\) Four `MSB4011` - https://github.com/dotnet/runtime/issues/70658
  - Introduced by the manual inclusion of `test_dependencies.*.props` in https://github.com/dotnet/runtime/pull/63978. There are now two NuGet-related `.props` that are almost identical. The batched build does its own restore and so isn't affected.
  - Upstream fix: https://github.com/dotnet/runtime/pull/80573.

2\) Two `IL2070`.
  - One DAM will solve the problem.

3\) `clang++ : warning : argument unused during compilation: '-I C:\Users\Accretion\source\dotnet\emsdk\upstream\emscripten\cache\sysroot\include\SDL'`.
  - Emscripten adds SDL headers by default, and this triggers the warnings (since, presumably, nothing references the code in them). We can pass `sUSE_SDL=0` ourselves to disable the default behavior, which could then go away once we update to emscripten which has https://github.com/emscripten-core/emscripten/pull/18443.

4\) `EXEC : warning : undefined symbol: S_P_CoreLib_Internal_Runtime_CompilerHelpers_ThrowHelpers__ThrowBodyRemoved` - appeared after the recent merge.
  - We are using the wrong helper for tentative instance method stubs.

5\) `EXEC : warning : undefined symbol: dotnet_browser_entropy` - good old `dotnet_browser_entropy`.
  - Presumably there because System.Native references it.

6\) `emcc : warning : warnings in JS library compilation [-Wjs-compiler]`.
  - Fallout from the undefined symbols warnings.

### PR DESCRIPTION
**Pull request by SingleAccretion**
Overall a pretty simple change:

1) Consider `pal_random.lib.js` an artifact of the `System.Native` build and pass it as `--js-library` to `emcc`.
2) Add a polyfill, since our Node doesn't support `crypto` directly (taken more or less verbatim from https://github.com/dotnet/runtime/pull/78696).

As a result, we now have a fully functional implementation and zero warnings in the runtime test build.

Fixes #2165.
Contributes to #2095.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

File: eng/liveBuilds.targets [hunk 1]
```diff
@@ -176,6 +176,10 @@
         $(LibrariesNativeArtifactsPath)*.pdb"
         IsNative="true"
         Exclude="@(ExcludeNativeLibrariesRuntimeFiles)" />
+      <LibrariesRuntimeFiles Condition="'$(TargetOS)' == 'Browser' and '$(RuntimeFlavor)' == 'CoreCLR'"
+                             Include="
+        $(LibrariesNativeArtifactsPath)pal_random.lib.js;"
```
Comment (yowl) [inline eng/liveBuilds.targets:181]: nit; this wrapping looks odd, is that intentional?

**Review by yowl — APPROVED** [review]
Nice, just the nit on the wrapping of the line.

File: eng/liveBuilds.targets (hunk 1 above)
Comment (SingleAccretion) [reply eng/liveBuilds.targets:181]: It is meant to be consistent with the rest of the file:
```
      <LibrariesRuntimeFiles Condition="'$(TargetOS)' == 'Browser' and '$(RuntimeFlavor)' == 'CoreCLR'"
                             Include="
        $(LibrariesNativeArtifactsPath)pal_random.lib.js;"
        IsNative="true" />
      <LibrariesRuntimeFiles Condition="'$(TargetOS)' == 'Browser' and '$(RuntimeFlavor)' == 'Mono'"
                             Include="
        $(LibrariesNativeArtifactsPath)dotnet.js;
        $(LibrariesNativeArtifactsPath)dotnet.d.ts;
        $(LibrariesNativeArtifactsPath)package.json;
        $(LibrariesNativeArtifactsPath)dotnet.wasm;
        $(LibrariesNativeArtifactsPath)dotnet.js.symbols;
        $(LibrariesNativeArtifactsPath)dotnet.timezones.blat;
        $(LibrariesNativeArtifactsPath)*.dat;"
        IsNative="true" />
```

Comment (yowl) [reply eng/liveBuilds.targets:181]: Good to go then, thanks

**Comment by jkotas** [comment]
Thank you!

**Pull request merged**